solr/core/src/java/org/apache/solr/security/SolrLogAuditLoggerPlugin.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.security;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Audit logger that writes to the Solr log
+ * @lucene.experimental
+ */
+public class SolrLogAuditLoggerPlugin extends AuditLoggerPlugin {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  /**
+   * Initialize the plugin from security.json
+   * @param pluginConfig the config for the plugin
+   */
+  @Override
+  public void init(Map<String, Object> pluginConfig) {
+    super.init(pluginConfig);
+    setFormatter(event ->
+        "type='" + event.getEventType().name() + '\'' +
```

Comment:
Should this be a StringBuilder? This will be done for every event, right?

Reply:
Agree